webapp/channels/src/components/convert_gm_to_channel_modal/convert_gm_to_channel_modal.tsx

```diff
@@ -0,0 +1,220 @@
+    const handleChannelURLChange = useCallback((newURL: string) => {
+        setChannelChannelURL(newURL);
+    }, []);
+
+    const [channelMemberNames, setChannelMemberNames] = useState<string[]>([]);
+
+    useEffect(() => {
+        const validProfilesInChannel = props.profilesInChannel.
+            filter((user) => user.id !== props.currentUserId && user.delete_at === 0).
+            map((user) => displayUsername(user, props.teammateNameDisplaySetting));
+
+        setChannelMemberNames(validProfilesInChannel);
+    }, [props.profilesInChannel]);
+
+    const [commonTeamsById, setCommonTeamsById] = useState<{[id: string]: Team}>({});
+    const [commonTeamsFetched, setCommonTeamsFetched] = useState<boolean>(false);
+    const [loadingAnimationTimeout, setLoadingAnimationTimeout] = useState<boolean>(false);
+    const [selectedTeamId, setSelectedTeamId] = useState<string>();
+    const [nameError, setNameError] = useState<boolean>(false);
+    const [conversionError, setConversionError] = useState<string>();
+
+    const handleTeamChange = useCallback((teamId: string) => {
+        setSelectedTeamId(teamId);
+    }, []);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const work = async () => {
+            const response = await dispatch(getGroupMessageMembersCommonTeams(props.channel.id)) as ActionResult<Team[], ServerError>;
+            if (response.error || !response.data) {
+                return;
+            }
+            const teams = response.data;
+
+            const teamsById: {[id: string]: Team} = {};
+            teams.forEach((team) => {
+                teamsById[team.id] = team;
+            });
+            setCommonTeamsById(teamsById);
+            setCommonTeamsFetched(true);
+
+            // if there is only common team, selected it.
+            if (teams.length === 1) {
+                setSelectedTeamId(teams[0].id);
+            }
+        };
+
+        work();
+        setTimeout(() => setLoadingAnimationTimeout(true), 1200);
+    }, []);
+
+    const handleConfirm = async () => {
+        if (!selectedTeamId) {
+            return;
+        }
+
+        const {error} = await props.actions.convertGroupMessageToPrivateChannel(props.channel.id, selectedTeamId, channelName.trim(), channelURL.trim());
+
+        if (error) {
+            setConversionError(error.message);
+            return;
+        }
+
+        setConversionError(undefined);
+        if (props.channelsCategoryId) {
+            props.actions.moveChannelsInSidebar(props.channelsCategoryId, 0, props.channel.id, false);
+        }
+        trackEvent('actions', 'convert_group_message_to_private_channel', {channel_id: props.channel.id});
+        props.onExited();
+    };
+
+    const showLoader = !commonTeamsFetched || !loadingAnimationTimeout;
+
+    const canCreate = (): boolean => {
+        return selectedTeamId !== undefined && channelName !== '' && !nameError;
+    };
```

Comment:
This is only used by being called in the render, so no need to be a function. A constant works the same.

Reply:
Done.